source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRViewController.mm

```diff
@@ -171,15 +178,36 @@ - (void)render
        [newView.topAnchor constraintEqualToAnchor:view.topAnchor]]];
 }
 
+auto _asyncRenderedElements = std::list<const BaseCardElement*>();
+
+- (void)callDidCompleteRenderingIfNeeded
+{
+#if DEBUG
+    NSLog(@"callDidCompleteRenderingIfNeeded");
+#endif
+
+    if (_asyncRenderedElements.size() == 0)
```

Comment:
consider this case, textblock is added, and completed. since _asyncRenderedElements.size() is zero, didCompleteRendering() will get called. but there were few more textblocks that were not put into dispatch queue because Image elements are being processed before it. If we have infinite queues, this won't be an issue, but I think there can be cases where this logic can fail since we don't have infinite queue.  I think you can add a additional check that makes sure we have seen all elements in the tree.